src/main/java/ch/njol/skript/SkriptCommand.java

```diff
@@ -398,60 +444,65 @@ public boolean onCommand(CommandSender sender, Command command, String label, St
 				htmlGenerator.generate(); // Try to generate docs... hopefully
 				Skript.info(sender, "All documentation generated!");
 			} else if (args[0].equalsIgnoreCase("test") && TestMode.DEV_MODE) {
-				File scriptFile;
+				File[] scriptFiles;
 				if (args.length == 1) {
-					scriptFile = TestMode.lastTestFile;
-					if (scriptFile == null) {
-						Skript.error(sender, "No test script has been run yet!");
+					scriptFiles = TestMode.lastTestFiles;
+					if (scriptFiles == null || scriptFiles.length == 0) {
+						Skript.error(sender, "No test script(s) have been ran yet!");
 						return true;
 					}
 				} else {
 					if (args[1].equalsIgnoreCase("all")) {
-						scriptFile = TestMode.TEST_DIR.toFile();
+						scriptFiles = new File[]{TestMode.TEST_DIR.toFile()};
 					} else {
-						scriptFile = getScriptFromArgs(sender, args, TestMode.TEST_DIR.toFile());
-						TestMode.lastTestFile = scriptFile;
+						scriptFiles = getScriptsFromArgs(sender, args, TestMode.TEST_DIR.toFile());
+						TestMode.lastTestFiles = scriptFiles;
 					}
 				}
 
-				if (scriptFile == null || !scriptFile.exists()) {
-					Skript.error(sender, "Test script doesn't exist!");
+				if (scriptFiles == null || scriptFiles.length == 0)
 					return true;
-				}
 
 				// Close previous loggers before we create a new one
 				// This prevents closing logger errors
 				timingLogHandler.close();
 				logHandler.close();
 
 				TestingLogHandler errorCounter = new TestingLogHandler(Level.SEVERE).start();
-				ScriptLoader.loadScripts(scriptFile, errorCounter)
-					.thenAccept(scriptInfo ->
-						// Code should run on server thread
-						Bukkit.getScheduler().scheduleSyncDelayedTask(Skript.getInstance(), () -> {
-							Bukkit.getPluginManager().callEvent(new SkriptTestEvent()); // Run it
-							ScriptLoader.unloadScripts(ScriptLoader.getLoadedScripts());
-
-							// Get results and show them
-							TestResults testResults = TestTracker.collectResults();
-							String[] lines = testResults.createReport().split("\n");
-							for (String line : lines) {
-								Skript.info(sender, line);
-							}
 
-							// Log results to file
-							Skript.info(sender, "Collecting results to " + TestMode.RESULTS_FILE);
-							String results = new GsonBuilder()
-								.setPrettyPrinting() // Easier to read lines
-								.disableHtmlEscaping() // Fixes issue with "'" character in test strings going unicode
-								.create().toJson(testResults);
-							try {
-								Files.writeString(TestMode.RESULTS_FILE, results);
-							} catch (IOException e) {
-								Skript.exception(e, "Failed to write test results.");
-							}
-						})
-					);
+				for (File scriptFile : scriptFiles) {
+					if (!scriptFile.exists()) {
+						Skript.error("The test script '" + scriptFile.getName() + "' doesn't exist!");
+						continue;
+					}
+					ScriptLoader.loadScripts(scriptFile, errorCounter)
```

Comment:
again about doing everything one by one instead of as a batch, especially since here each subsequent script will overwrite the previous results in the result file.